stock_picking_portal/controllers/portal.py

```diff
@@ -0,0 +1,273 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import binascii
+from collections import OrderedDict
+
+from odoo import _, fields, http
+from odoo.exceptions import AccessDenied, AccessError, MissingError, UserError
+from odoo.http import request
+
+from odoo.addons.portal.controllers import portal
+from odoo.addons.portal.controllers.mail import _message_post_helper
+from odoo.addons.portal.controllers.portal import pager as portal_pager
+
+
+class CustomerPortal(portal.CustomerPortal):
+    def _get_prepared_operation_domain(self, partner):
+        """Returns a domain to search for operations for the given partner.
+
+        Args:
+            partner (res.partner): partner to search operations for.
+
+        Returns:
+            list: domain to search for operations for the given partner.
+
+        """
+        StockPicking = request.env["stock.picking"]
+        portal_visible_operation_ids = StockPicking._get_available_operations()
+        return [
+            ("partner_id", "=", partner.id),
+            ("picking_type_id", "in", portal_visible_operation_ids),
+        ]
+
+    def _prepare_home_portal_values(self, counters):
+        values = super()._prepare_home_portal_values(counters)
+        partner = request.env.user.partner_id
+        StockPicking = request.env["stock.picking"]
+        domain = self._get_prepared_operation_domain(partner)
+        stock_operations_count = StockPicking.search_count(domain)
+        if "stock_operations_count" in counters:
+            values["stock_operations_count"] = (
+                stock_operations_count if stock_operations_count > 0 else "0"
+            )
+        return values
+
+    @http.route(
+        ["/my/stock_operations", "/my/stock_operations/page/<int:page>"],
+        type="http",
+        auth="user",
+        website=True,
+    )
+    def portal_my_stock_operations(self, **kwargs):
```

Comment:
Please add doc string for this endpoint.

Reply:
Thanks, I'll add it